src/libYARP_sig/src/IntrinsicParams.cpp

```diff
@@ -12,7 +12,7 @@
 using namespace yarp::sig;
 
 IntrinsicParams::IntrinsicParams(): principalPointX(0.0), principalPointY(0.0),
-                   focalLengthX(0.0), focalLengthY(0.0),
+                   physFocalLength(0.0), focalLengthX(0.0), focalLengthY(0.0),
```

Comment:
I'd set `nan` as a initialization so that if the data is not set by the device implementation, it is clear that the info is not usable.
(btw, maybe it'd be a better initialization for all of them)

Reply:
I'm not sure `nan` is a good thing for a well defined behavior of C++ code, as very few programmers check if a value is `nan` before using it.

On the opposite I think a common check for length parameters is to verify if they are strictly positive, for example with something like `if(focalLength <= 0) return error;`.
Using a `nan` would break this check because `nan <= 0` is always `false`.